tests/trout/models.py

```diff
@@ -561,6 +566,17 @@ def main() -> None:
     if comment.null_str_specified is not None:
         print(comment.null_str_specified)
 
+    process_non_nullable(comment.other_metadata)
+    if isinstance(comment.other_metadata_nullable, type(None)):
+        print()
+    # refinement doesn't work
+    # see: https://github.com/python/mypy/issues/9783
```

Comment:
this is cool with me as long as this isn't worse than the existing behavior for mypy users

Reply:
I actually found today that comment you've made in another case very similar.

So, yes, this should be _a lot better_ than the existing behaviour. The existing one was `Any` always and without a way to make it better. In the current one you can pass something to it and it will be validated like that, for both __get__ and __set__.

You will notice that the `ArrayField` have a similar issue, but its typing is greatly improved. E.g. `ArrayField(ArrayField(CharField))` will be typed and validated as `List[List[str]]`.

Also, note that like I described in the comment, both mypy and pyright identifies the field correctly. mypy will say that this is a  `'builtins.dict*[builtins.str, builtins.list[builtins.int]]'`, so it is indeed understanding the typing (pyright says `Dict[str, List[str]]`). The reason it didn't mark this as unreachable is probably due to the issue you opened.